### PR TITLE
Commit for adding-devise

### PR DESCRIPTION
**Pull request by jessicatitus**
We set up devise, created a user model and migration, made a custom email validator to check for .com and .net, and wrote tests and passed them.

### CODE REVIEW

**Comment by jessicatitus** [comment]
Passing and ready!!

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_filter :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:first_name, :last_name])
```
Comment (chrissyfancy) [inline app/controllers/application_controller.rb:8]: This looks great! Be sure to add any additional fields that you want to be passed into the user object on sign up here.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true, email: true
+
```
Comment (chrissyfancy) [inline app/models/user.rb:5]: This is fine for now, since devise has built in validations for these fields.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  it 'has a matching password confirmation for the password' do
+    user = User.new
+    user.password = 'password'
+    user.password_confirmation = 'anotherpassword'
+
+    expect(user).to_not be_valid
+    expect(user.errors[:password_confirmation]).to_not be_blank
+  end
+end
```
Comment (chrissyfancy) [inline spec/models/user_spec.rb:22]: Nice job with your testing!

**Pull request merged**